### PR TITLE
Switch to thiserror for error handling

### PR DESCRIPTION
**Pull request by LoganDark**
`failure` is a legacy, **now-deprecated** crate that is not much use nowadays since the only way for it to interoperate with other crates is to call `.compat` on the `Fail`, which is kind of... well... a total fail on their part.

I recently went through a lot of frustration trying to convert `failure::Error` to anything else, since it does not implement any of the error traits, and only seems to provide weird metadata like backtraces and contexts instead of a user-friendly error message. (I suppose it provides an error message if you call `to_string()` on it.)

All in all, I think this change is for the better, and since the major version is still 0, minor breaking changes like this one should be fine. This PR will be version 0.1.0. `thiserror` works much more nicely with other crates like `anyhow`.

All tests still pass.

### CODE REVIEW

**Comment by LoganDark** [comment]
Example of code before this PR, using Anyhow as an example:

```rust
client.execute(Account::get())
		.map_err(|e| anyhow::Error::from(e.compat()))
		.context("testing DigitalOcean key")?;
```

Example of code after this PR:

```rust
client.execute(Account::get())
		.context("testing DigitalOcean key")?;
```

In my opinion, `thiserror` properly integrates into Rust's error system (and `anyhow` picks up on that), but `failure` tries to do its own thing and ends up being a frustration point.

**Comment by Hoverbear** [comment]
Oh my gosh, yay!

**Comment by Hoverbear** [comment]
It seems I need to repair the CI here. :( This repo has kind of fallen dormant.

I'd love to merge this though. Let me test it locally.

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "digitalocean"
-version = "0.0.4"
+version = "0.1.0"
```
Comment (Hoverbear) [inline Cargo.toml:3]: Okay fine. :)

**Comment by LoganDark** [comment]
If you need someone to help maintain this library, I'm game.

**Comment by Hoverbear** [comment]
@LoganDark I'd really love someone to hack with on this! If you want to contribute I'd love to be at least a responsible reviewer, but I'm sure I'll get inspired... You're very welcome to also join this chat and talk about it with me! https://discord.gg/cfx3mHj

**Comment by Hoverbear** [comment]
Local tests are green, merging.

**Pull request merged**